### PR TITLE
Slurm in azhpc

### PR DESCRIPTION
**Pull request by edwardsp**
This is a first version but a working example.  Still more work to do including:

* Tidying up where to install
* Multiple queues
* Supporting more VM types

### CODE REVIEW

File: examples/slurm_autoscale/scripts/hpcadmin_id_rsa [hunk 1]
```diff
@@ -0,0 +1 @@
+../hpcadmin_id_rsa
```
Comment (xpillons) [inline examples/slurm_autoscale/scripts/hpcadmin_id_rsa:1]: probably this is not needed

File: examples/slurm_autoscale/scripts/hpcadmin_id_rsa.pub [hunk 2]
```diff
@@ -0,0 +1 @@
+../hpcadmin_id_rsa.pub
```
Comment (xpillons) [inline examples/slurm_autoscale/scripts/hpcadmin_id_rsa.pub:1]: probably this is not needed

File: examples/slurm_autoscale/scripts/munge.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (xpillons) [inline examples/slurm_autoscale/scripts/munge.sh:1]: can all these scripts be moved to the azhpc/scripts directory ?

File: examples/slurm_autoscale/scripts/config.json [hunk 4]
```diff
@@ -0,0 +1 @@
+../config.json
```
[the remark this answers is not in the document]
Comment (edwardsp) [reply examples/slurm_autoscale/scripts/config.json:1]: they are symbolic links in order to copy to the headnode

File: examples/slurm_autoscale/scripts/hpcadmin_id_rsa (hunk 1 above)
Comment (edwardsp) [reply examples/slurm_autoscale/scripts/hpcadmin_id_rsa:1]: see above

File: examples/slurm_autoscale/scripts/hpcadmin_id_rsa.pub (hunk 2 above)
Comment (edwardsp) [reply examples/slurm_autoscale/scripts/hpcadmin_id_rsa.pub:1]: see above

**Pull request merged**